Classes/FairPlayDRM/FPSLicenseHelper.swift

```diff
@@ -47,36 +96,24 @@ class FPSLicenseHelper {
     func performCKCRequest(_ spcData: Data, url: URL, callback: @escaping (FPSLicense?, Error?) -> Void) {
         
         
-        var requestParams = PKRequestParams(url: url, headers: ["Content-Type": "application/octet-stream"])
-
+        var requestParams = PKRequestParams(url: url, headers: nil)
+        
         if let adapter = self.params?.requestAdapter {
             requestParams = adapter.adapt(requestParams: requestParams)
         }
         
-        var request = URLRequest(url: requestParams.url)
-        if let headers = requestParams.headers {
-            for (header, value) in headers {
-                request.setValue(value, forHTTPHeaderField: header)
-            }
-        }
+        let licenseProvider = self.params?.licenseProvider ?? KalturaFairPlayLicenseProvider.sharedInstance
 
-        request.httpBody = spcData.base64EncodedData()
-        request.httpMethod = "POST"
-        
-        PKLog.debug("Sending SPC to server")
-        let startTime = Date.timeIntervalSinceReferenceDate
-        let dataTask = URLSession.shared.dataTask(with: request) { (data: Data?, response: URLResponse?, error: Error?) -> Void in
-            do {
-                let endTime: Double = Date.timeIntervalSinceReferenceDate
-                PKLog.debug("Got response in \(endTime-startTime) sec")
-                let lic = try FPSLicense(jsonResponse: data)
-                callback(lic, nil)
-                
-            } catch let e {
-                callback(nil, e)
-            }
+        licenseProvider.getLicense(spc: spcData, 
+                                               requestParams: requestParams) { (ckc, duration, error) in
+                                                
+                                                guard let ckc = ckc else {
```

Comment:
should we also check the duration here? >0?

Reply:
This duration is only for persistence, it is allowed to be 0.